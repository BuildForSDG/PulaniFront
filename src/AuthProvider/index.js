const authProvider = {
  login: ({ username, password }) => {
    const request = new Request('https://pulani2.herokuapp.com', {
      method: 'POST',
      body: JSON.stringify({ username, password }),
      headers: new Headers({ 'Content-Type': 'application/json' })
    });
    return fetch(request)
      .then((response) => {
        if (response.status < 200 || response.status >= 300) {
          throw new Error(response.statusText);
        }
        return response.json();
      })
      .then(({ token }) => {
        localStorage.setItem('token', token);
      });
  },
  // ...
  logout: () => Promise.resolve(),
  checkAuth: () => Promise.resolve(),
  checkError: () => Promise.resolve(),
  getPermissions: () => Promise.resolve()
};

export default authProvider;
