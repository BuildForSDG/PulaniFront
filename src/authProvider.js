// handle authentication
const authProvider = {
  login: ({ phone, password }) => {
    const url = process.env.REACT_APP_HEROKU + '/auth/login';

    const request = new Request(url, {
      method: 'POST',
      body: JSON.stringify({ phone, password }),
      headers: new Headers({ 'Content-Type': 'application/json' })
    });
    return fetch(request)
      .then(response => {
        if (response.status < 200 || response.status >= 300) {
          throw new Error('Wrong information. Please try again!');
        }
        return response.json();
      })
      .then(({ token }) => {
        sessionStorage.setItem('token', token);
      });
  },
  logout: () => {
    sessionStorage.removeItem('token');
    return Promise.resolve();
  },
  checkError: (error) => {
    const status = error.status;
    if (status === 401 || status === 403) {
      sessionStorage.removeItem('token');
      return Promise.reject();
    }
    return Promise.resolve();
  },
  checkAuth: () => {
    // check for a token
    return sessionStorage.getItem('token') ? Promise.resolve() : Promise.reject();
  },
  getPermissions: () => Promise.resolve()
};

export default authProvider;
