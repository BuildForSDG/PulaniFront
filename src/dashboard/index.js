import React from 'react';
import format from 'date-fns/format';
import ChartistGraph from 'react-chartist';
import clsx from 'clsx';
import Icon from '@material-ui/core/Icon';
import AttachMoney from '@material-ui/icons/AttachMoney';
import Store from '@material-ui/icons/Store';
import DateRange from '@material-ui/icons/DateRange';
import LocalOffer from '@material-ui/icons/LocalOffer';
import Update from '@material-ui/icons/Update';
import ArrowUpward from '@material-ui/icons/ArrowUpward';
import AccessTime from '@material-ui/icons/AccessTime';

import { 
  Card,
  CardIcon,
  CardHeader,
  CardBody,
  CardFooter,
  GridContainer, 
  GridItem,
  // charts
  dailySalesChart,
  savingsTrendsChart,
  budgetTrackerChart  
} from '../layout';

import { dashboardStyle } from '../assets/jss';

export const Dashboard = () => {
  const classes = dashboardStyle();

  let firstName = sessionStorage.getItem('firstName'),
       lastName = sessionStorage.getItem('lastName');  

  return (
    <div>
      <GridContainer>
        <GridItem xs={12}>
          <Card>
            <CardBody color='success'>
              <h2 className={classes.cardTitle}>
                Hi, {firstName}&nbsp;{lastName} 
                <span className={clsx('date')}>{format(new Date(), 'MMMM, DD')}</span>
              </h2>
            </CardBody>
          </Card>
        </GridItem>
      </GridContainer>
      <GridContainer>
        <GridItem xs={12} sm={6} md={4}>
          <Card>
            <CardHeader color='success' stats icon>
              <CardIcon color='success'>
                <AttachMoney />
              </CardIcon>
              <p className={classes.cardCategory}>Income Goal</p>
              <h3 className={classes.cardTitle}>UGX. 50,000</h3>
            </CardHeader>
            <CardFooter stats>
              <div className={classes.stats}>
                <Update />
                Daily Income
              </div>
            </CardFooter>
          </Card>
        </GridItem>
        <GridItem xs={12} sm={6} md={4}>
          <Card>
            <CardHeader color='warning' stats icon>
              <CardIcon color='warning'>
                <Icon>content_copy</Icon>
              </CardIcon>
              <p className={classes.cardCategory}>Savings Target</p>
              <h3 className={classes.cardTitle}>UGX. 500,000</h3>
            </CardHeader>
            <CardFooter stats>
              <div className={classes.stats}>
                <LocalOffer />
                Annual Savings
              </div>
            </CardFooter>
          </Card>
        </GridItem>
        <GridItem xs={12} sm={6} md={4}>
          <Card>
            <CardHeader color='danger' stats icon>
              <CardIcon color='danger'>
                <Store />
              </CardIcon>
              <p className={classes.cardCategory}>Budget Limit</p>
              <h3 className={classes.cardTitle}>UGX. 75,000</h3>
            </CardHeader>
            <CardFooter stats>
              <div className={classes.stats}>
                <DateRange />
                Monthly Budget
              </div>
            </CardFooter>
          </Card>
        </GridItem>
      </GridContainer>
      <GridContainer>
        <GridItem xs={12} sm={12} md={4}>
          <Card chart>
            <CardHeader color='success'>
              <ChartistGraph
                className='ct-chart'
                data={dailySalesChart.data}
                type='Line'
                options={dailySalesChart.options}
                listener={dailySalesChart.animation}
              />
            </CardHeader>
            <CardBody>
              <h4 className={classes.cardTitle}>Your daily sales</h4>
              <p className={classes.cardCategory}>
                <span className={classes.successText}>
                  <ArrowUpward className={classes.upArrowCardCategory} /> 55%
                </span>{' '}
                increase in today sales.
              </p>
            </CardBody>
            <CardFooter chart>
              <div className={classes.stats}>
                <AccessTime /> updated 4 minutes ago
              </div>
            </CardFooter>
          </Card>
        </GridItem>
        <GridItem xs={12} sm={12} md={4}>
          <Card chart>
            <CardHeader color='warning'>
              <ChartistGraph
                className='ct-chart'
                data={savingsTrendsChart.data}
                type='Bar'
                options={savingsTrendsChart.options}
                responsiveOptions={savingsTrendsChart.responsiveOptions}
                listener={savingsTrendsChart.animation}
              />
            </CardHeader>
            <CardBody>
              <h4 className={classes.cardTitle}>Your savings trends</h4>
              <p className={classes.cardCategory}>last year&rsquo;s performance</p>
            </CardBody>
            <CardFooter chart>
              <div className={classes.stats}>
                <AccessTime /> updated last year
              </div>
            </CardFooter>
          </Card>
        </GridItem>
        <GridItem xs={12} sm={12} md={4}>
          <Card chart>
            <CardHeader color='danger'>
              <ChartistGraph
                className='ct-chart'
                data={budgetTrackerChart.data}
                type='Line'
                options={budgetTrackerChart.options}
                listener={budgetTrackerChart.animation}
              />
            </CardHeader>
            <CardBody>
              <h4 className={classes.cardTitle}>Your tracked expenses</h4>
              <p className={classes.cardCategory}>last month trends</p>
            </CardBody>
            <CardFooter chart>
              <div className={classes.stats}>
                <AccessTime /> updated last week
              </div>
            </CardFooter>
          </Card>
        </GridItem>
      </GridContainer>
    </div>
  );
}
