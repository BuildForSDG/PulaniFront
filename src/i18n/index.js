import polyglotI18nProvider from 'ra-i18n-polyglot';
import englishMessages from './en';

const i18nProvider = polyglotI18nProvider(
  locale => {
    if (locale === 'fr') {
      // Plan to include translation for local languages in Uganda
      return import('./fr').then(messages => messages.default);
    }
    // Always fallback on english
    return englishMessages;
  },
  'en',
  { allowMissing: true }
);

export default i18nProvider;
