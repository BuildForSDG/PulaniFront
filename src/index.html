<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta name="description" content="Get the most out of your day" />
    <title>Pulani</title>
  </head>
  <body>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root" class="pulani"></div>
    <script src="./index.js"></script>
  </body>
</html>
