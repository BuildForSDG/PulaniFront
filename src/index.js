import 'react-app-polyfill/ie11';
import 'react-app-polyfill/stable';
import 'proxy-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
import * as serviceWorker from './serviceWorker';
import './stylesheets/index.css';

ReactDOM.render(<App/>, document.getElementById('root'));
serviceWorker.register();
