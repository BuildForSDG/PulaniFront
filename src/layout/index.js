export * from './AppBar';
export * from './Layout';

export * from './components';
