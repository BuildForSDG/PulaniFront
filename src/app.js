import React from 'react';
import { Admin, fetchUtils, Resource } from 'react-admin';

import dataProvider from './dataProvider';
import { API, authProvider } from './authProvider';
import i18nProvider from './i18n';
import { PulaniLayout } from './layout';
import { PulaniLoginPage } from './login';
import { Dashboard } from './dashboard';

const App = () => {
  const [data, setData] = React.useState(null);

  const fetchDataProvider = async () => {
    const httpClient = (url, options) => {
      if (!options.headers) {
        options.headers = new Headers({ Accept: 'application/json' });
      }
      const token = sessionStorage.getItem('token');
      options.headers.set('Authorization', `Bearer ${token}`);
      return fetchUtils.fetchJson(url, options);
    };

    let dataProviderInstance = await dataProvider(API, httpClient);

    setData(
      // NOTE: dataProviderInstance can be a function
      () => dataProviderInstance
    );
  };

  React.useEffect(() => {
    fetchDataProvider();
    // cleanup
    return () => fetchDataProvider();
  }, []);

  if (!data) {
    return (
      <div className="loader-container">
        <div className="loader">Loading...</div>
      </div>
    );
  }
  
  return (
    <Admin
      title=""
      dataProvider={data}
      authProvider={authProvider}
      i18nProvider={i18nProvider}
      layout={PulaniLayout}
      loginPage={PulaniLoginPage}
      dashboard={Dashboard}
    >
      <Resource name="users" />
    </Admin>
  );
};

export default App;
