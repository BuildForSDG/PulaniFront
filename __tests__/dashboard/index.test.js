import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from '../../src/dashboard';

describe('<Dashboard/> module', () => {
	const dashboard = shallow(<Dashboard />);

	it('that it exists', () => {
		expect(dashboard).toBeDefined();
	});

	it('that it renders without crashing', () => {
		expect(dashboard).toMatchSnapshot();
	});
});
