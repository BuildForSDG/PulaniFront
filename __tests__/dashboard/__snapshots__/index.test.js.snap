// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> module that it renders without crashing 1`] = `
<div>
  <GridContainer>
    <GridItem
      xs={12}
    >
      <Card>
        <CardBody
          color="success"
        >
          <h2
            className="PulaniDashboard-cardTitle-6"
          >
            Hi, 
             
            <span
              className="date"
            >
              June, 10
            </span>
          </h2>
        </CardBody>
      </Card>
    </GridItem>
  </GridContainer>
  <GridContainer>
    <GridItem
      md={4}
      sm={6}
      xs={12}
    >
      <Card>
        <CardHeader
          color="success"
          icon={true}
          stats={true}
        >
          <CardIcon
            color="success"
          >
            <AttachMoneyIcon />
          </CardIcon>
          <p
            className="PulaniDashboard-cardCategory-4"
          >
            Income Goal
          </p>
          <h3
            className="PulaniDashboard-cardTitle-6"
          >
            UGX. 50,000
          </h3>
        </CardHeader>
        <CardFooter
          stats={true}
        >
          <div
            className="PulaniDashboard-stats-3"
          >
            <UpdateIcon />
            Daily Income
          </div>
        </CardFooter>
      </Card>
    </GridItem>
    <GridItem
      md={4}
      sm={6}
      xs={12}
    >
      <Card>
        <CardHeader
          color="warning"
          icon={true}
          stats={true}
        >
          <CardIcon
            color="warning"
          >
            <WithStyles(ForwardRef(Icon))>
              content_copy
            </WithStyles(ForwardRef(Icon))>
          </CardIcon>
          <p
            className="PulaniDashboard-cardCategory-4"
          >
            Savings Target
          </p>
          <h3
            className="PulaniDashboard-cardTitle-6"
          >
            UGX. 500,000
          </h3>
        </CardHeader>
        <CardFooter
          stats={true}
        >
          <div
            className="PulaniDashboard-stats-3"
          >
            <LocalOfferIcon />
            Annual Savings
          </div>
        </CardFooter>
      </Card>
    </GridItem>
    <GridItem
      md={4}
      sm={6}
      xs={12}
    >
      <Card>
        <CardHeader
          color="danger"
          icon={true}
          stats={true}
        >
          <CardIcon
            color="danger"
          >
            <StoreIcon />
          </CardIcon>
          <p
            className="PulaniDashboard-cardCategory-4"
          >
            Budget Limit
          </p>
          <h3
            className="PulaniDashboard-cardTitle-6"
          >
            UGX. 75,000
          </h3>
        </CardHeader>
        <CardFooter
          stats={true}
        >
          <div
            className="PulaniDashboard-stats-3"
          >
            <DateRangeIcon />
            Monthly Budget
          </div>
        </CardFooter>
      </Card>
    </GridItem>
  </GridContainer>
  <GridContainer>
    <GridItem
      md={4}
      sm={12}
      xs={12}
    >
      <Card
        chart={true}
      >
        <CardHeader
          color="success"
        >
          <ChartistGraph
            className="ct-chart"
            data={
              Object {
                "labels": Array [
                  "M",
                  "T",
                  "W",
                  "T",
                  "F",
                  "S",
                  "S",
                ],
                "series": Array [
                  Array [
                    12,
                    17,
                    7,
                    17,
                    23,
                    18,
                    38,
                  ],
                ],
              }
            }
            listener={
              Object {
                "draw": [Function],
              }
            }
            options={
              Object {
                "chartPadding": Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                  "top": 0,
                },
                "high": 50,
                "lineSmooth": [Function],
                "low": 0,
              }
            }
            type="Line"
          />
        </CardHeader>
        <CardBody>
          <h4
            className="PulaniDashboard-cardTitle-6"
          >
            Your daily sales
          </h4>
          <p
            className="PulaniDashboard-cardCategory-4"
          >
            <span
              className="PulaniDashboard-successText-1"
            >
              <ArrowUpwardIcon
                className="PulaniDashboard-upArrowCardCategory-2"
              />
               55%
            </span>
             
            increase in today sales.
          </p>
        </CardBody>
        <CardFooter
          chart={true}
        >
          <div
            className="PulaniDashboard-stats-3"
          >
            <AccessTimeIcon />
             updated 4 minutes ago
          </div>
        </CardFooter>
      </Card>
    </GridItem>
    <GridItem
      md={4}
      sm={12}
      xs={12}
    >
      <Card
        chart={true}
      >
        <CardHeader
          color="warning"
        >
          <ChartistGraph
            className="ct-chart"
            data={
              Object {
                "labels": Array [
                  "Jan",
                  "Feb",
                  "Mar",
                  "Apr",
                  "Mai",
                  "Jun",
                  "Jul",
                  "Aug",
                  "Sep",
                  "Oct",
                  "Nov",
                  "Dec",
                ],
                "series": Array [
                  Array [
                    42,
                    85,
                    85,
                    85,
                    138,
                    181,
                    181,
                    215,
                    215,
                    225,
                    281,
                    366,
                  ],
                ],
              }
            }
            listener={
              Object {
                "draw": [Function],
              }
            }
            options={
              Object {
                "axisX": Object {
                  "showGrid": false,
                },
                "chartPadding": Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 5,
                  "top": 0,
                },
                "high": 500,
                "low": 0,
              }
            }
            responsiveOptions={
              Array [
                Array [
                  "screen and (max-width: 640px)",
                  Object {
                    "axisX": Object {
                      "labelInterpolationFnc": [Function],
                    },
                    "seriesBarDistance": 5,
                  },
                ],
              ]
            }
            type="Bar"
          />
        </CardHeader>
        <CardBody>
          <h4
            className="PulaniDashboard-cardTitle-6"
          >
            Your savings trends
          </h4>
          <p
            className="PulaniDashboard-cardCategory-4"
          >
            last year’s performance
          </p>
        </CardBody>
        <CardFooter
          chart={true}
        >
          <div
            className="PulaniDashboard-stats-3"
          >
            <AccessTimeIcon />
             updated last year
          </div>
        </CardFooter>
      </Card>
    </GridItem>
    <GridItem
      md={4}
      sm={12}
      xs={12}
    >
      <Card
        chart={true}
      >
        <CardHeader
          color="danger"
        >
          <ChartistGraph
            className="ct-chart"
            data={
              Object {
                "labels": Array [
                  "Week 1",
                  "Week 2",
                  "Week 3",
                  "Week 4",
                ],
                "series": Array [
                  Array [
                    20,
                    35,
                    55,
                    40,
                  ],
                ],
              }
            }
            listener={
              Object {
                "draw": [Function],
              }
            }
            options={
              Object {
                "chartPadding": Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                  "top": 0,
                },
                "high": 75,
                "lineSmooth": [Function],
                "low": 0,
              }
            }
            type="Line"
          />
        </CardHeader>
        <CardBody>
          <h4
            className="PulaniDashboard-cardTitle-6"
          >
            Your tracked expenses
          </h4>
          <p
            className="PulaniDashboard-cardCategory-4"
          >
            last month trends
          </p>
        </CardBody>
        <CardFooter
          chart={true}
        >
          <div
            className="PulaniDashboard-stats-3"
          >
            <AccessTimeIcon />
             updated last week
          </div>
        </CardFooter>
      </Card>
    </GridItem>
  </GridContainer>
</div>
`;
